pkg/manifests/config.go

```diff
@@ -198,6 +203,72 @@ func (c *Config) SetImages(images map[string]string) {
 	c.Images.K8sPrometheusAdapter = images["k8s-prometheus-adapter"]
 }
 
+func (c *Config) LoadClusterID(load func() (*configv1.ClusterVersion, error)) error {
+	if c.TelemeterClientConfig.ClusterID != "" {
+		return nil
+	}
+
+	cv, err := load()
+	if err != nil {
+		return fmt.Errorf("error loading cluster version: %v", err)
+	}
+
+	c.TelemeterClientConfig.ClusterID = string(cv.Spec.ClusterID)
+	return nil
+}
+
+func (c *Config) LoadToken(load func() (*v1.ConfigMap, error)) error {
+	if c.TelemeterClientConfig.Token != "" {
+		return nil
+	}
+
+	cmap, err := load()
+	if err != nil {
+		return fmt.Errorf("error loading configmap: %v", err)
+	}
+
+	ic := make(map[string]interface{})
+	if err := yaml.Unmarshal([]byte(cmap.Data["install-config"]), &ic); err != nil {
+		return fmt.Errorf("unmarshaling install config failed: %v", err)
+	}
+
+	if _, ok := ic["pullSecret"].(string); !ok {
```

Comment:
hah! very good catch, this is a regression indeed! casting to a string from a nil interface definitely panics :+1: 
https://play.golang.org/p/SKjzaoPgRNi

Reply:
oh, wait! the `ok` catches it ;-)

Reply:
https://play.golang.org/p/9Qo9B3nHxEs

Reply:
I'm talking about the case where pullSecret doesn't exist https://play.golang.org/p/PNzSGVyZlxb

Reply:
sorry, if I'm blind, but since it is guarded, it shouldn't panic, right?
https://play.golang.org/p/3-4Eyu19Yfh

nevermind, I'll add unit tests here ;-)